module/gui/connector.py

```diff
@@ -67,35 +71,163 @@ def connectProxy(self):
+        soc.close()
+        if gaierror:
+            if firstAttempt:
+                return False
+            self.messageBox_01(self.host, self.port)
+            return False
+        if timeout:
+            if firstAttempt:
+                return False
+            if not self.messageBox_02(self.host, self.port):
+                return False
+            else:
+                self.timeoutTimerStart()
+        # login
+        while True:
+            err = None
+            errlogin = False
+            try:
+                client = ThriftClient(self.host, self.port, self.user, self.password)
+            except WrongLogin:
+                errlogin = True
+            except NoSSL:
+                err = "nossl"
+            except NoConnection:
+                err = "noconn"
+            
+            if not errlogin:
+                break
+            
+            # user and password popup
+            if self.messageBox_03(self.host, self.port, self.user, self.password) == QDialog.Rejected:
+                return False
+            else:
+                self.timeoutTimerStart()
+            self.user = unicode(self.pwBox.userLE.text())
+            self.password = unicode(self.pwBox.passwordLE.text())
+            sleep(1) # some delay to let the dialog fade out
+        
+        if err is not None:
+            if firstAttempt:
+                return False
+            if err == "nossl":
+                self.messageBox_04(self.host, self.port)
+            elif err == "noconn":
+                self.messageBox_05(self.host, self.port)
             return False
         
+        self.ssl = client.isSSLConnection() # remember if we are connected with SSL
         self.proxy = DispatchRPC(self.mutex, client)
         self.connect(self.proxy, SIGNAL("connectionLost"), self, SIGNAL("connectionLost"))
         
+        # check server version
         server_version = self.proxy.getServerVersion()
         self.connectionID = uuid().hex
-        
         if not server_version == SERVER_VERSION:
-            self.emit(SIGNAL("errorBox"), _("server is version %(new)s client accepts version %(current)s") % { "new": server_version, "current": SERVER_VERSION})
+            if firstAttempt:
+                return False
+            self.messageBox_06(server_version, self.host, self.port)
             return False
         
         return True
     
+    def isSSLConnection(self):
+        if self.internal:
+            return False
+        return self.ssl
+    
+    def getOurUserData(self):
+        return self.proxy.getUserData(self.user, self.password)
+    
+    def disconnectProxy(self):
+        """
+            close the sockets
+        """
+        if self.internal:
+            return
+        self.proxy.server.close()
+        self.proxy = self.Dummy()
+    
+    def timeoutTimerStart(self):
+        self.emit(SIGNAL("connectTimeout"))
```

Comment:
![Codacy](https://app.codacy.com/assets/images/favicon.png) Issue found: ['SIGNAL' may be undefined, or defined from star imports: PyQt4.QtCore, PyQt4.QtGui](https://app.codacy.com/manual/pyLoad/pyload/pullRequest?prid=567903)